sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Filter.java

```diff
@@ -207,43 +166,54 @@ public void populateDisplayData(DisplayData.Builder builder) {
    * <p>See also {@link #byPredicate}, which returns elements
    * that satisfy the given predicate.
    */
-  public static <T extends Comparable<T>> ParDo.Bound<T, T> greaterThanEq(final T value) {
-    return ParDo.named("Filter.greaterThanEq").of(new DoFn<T, T>() {
-      @Override
-      public void processElement(ProcessContext c) {
-        if (c.element().compareTo(value) >= 0) {
-          c.output(c.element());
-        }
-      }
-
+  public static <T extends Comparable<T>> Filter<T> greaterThanEq(final T value) {
+    return byPredicate(new SerializableFunction<T, Boolean>() {
       @Override
-      public void populateDisplayData(DisplayData.Builder builder) {
-        super.populateDisplayData(builder);
-        Filter.populateDisplayData(builder, String.format("x ≥ %s", value));
+      public Boolean apply(T input) {
+        return input.compareTo(value) >= 0;
       }
-    });
+    }).named("Filter.greaterThanEq").described(String.format("x ≥ %s", value));
   }
 
   ///////////////////////////////////////////////////////////////////////////////
 
+  private String name = "Filter";
   private SerializableFunction<T, Boolean> predicate;
+  private String predicateDescription = "Filter.predicate";
 
   private Filter(SerializableFunction<T, Boolean> predicate) {
     this.predicate = predicate;
   }
 
-  private Filter(String name, SerializableFunction<T, Boolean> predicate) {
-    super(name);
+  private Filter(String name, SerializableFunction<T, Boolean> predicate,
+      String predicateDescription) {
+    this.name = name;
     this.predicate = predicate;
+    this.predicateDescription = predicateDescription;
   }
 
+  /**
+   * Returns a new {@link Filter} {@link PTransform} that's like this
+   * {@link PTransform} but with the specified name. Does not
+   * modify this {@link PTransform}.
+   */
   public Filter<T> named(String name) {
-    return new Filter<>(name, predicate);
+    return new Filter<>(name, predicate, predicateDescription);
+  }
+
+  /**
+   * Returns a new {@link Filter} {@link PTransform} that's like this
+   * {@link PTransform} but with the specified description for {@link DisplayData}. Does not
+   * modify this {@link PTransform}.
+   */
+  public Filter<T> described(String description) {
```

Comment:
Recommend removing this from the public API as it doesn't add anything beyond what a user could accomplish beyond `.named(String)`. Simply making this package-private should be sufficient.

If there's a use-case for adding display data for user-defined filters, we would probably want to add the ability to implement `HasDisplayData` on the predicate.